docs: Update gateway-api version to v0.6.1

sayboras opened the pull request:
After PR #22680, the minimum Gateway API supported version is v0.6.x. The older version will not work due to changes in the API version of CRDs.

Relates: #22680
Fixes: #25377

sayboras committed:
docs: Update gateway-api version to v0.6.1

After the PR #22680, the minimum Gateway API supported version is
v0.6.x. Older version will not work due to changes in API version of
CRDs.

Relates: #22680
Fixes: #25377

Signed-off-by: Tam Mach <tam.mach@cilium.io>

raphink commented on Documentation/network/servicemesh/gateway-api/gateway-api.rst:
```diff
@@ -33,4 +33,4 @@ Cilium's Gateway API features:
 
-More examples can be found `upstream repository <https://github.com/kubernetes-sigs/gateway-api/tree/v0.5.1/examples/v1beta1>`_.
\ No newline at end of file
+More examples can be found `upstream repository <https://github.com/kubernetes-sigs/gateway-api/tree/v0.6.1/examples/v1beta1>`_.
```
This path does not seem to exist

sayboras replied:
Thanks, I have updated the path :+1:

sayboras committed:
docs: Add upgrade docs for Gateway API

Without ReferenceGrant v1beta1, we will have below error in the operator
upon starting up. The existing provisioned resource will continue to work
as usual, but new resources will not be handled properly.

```
2023-05-10T13:49:48.691595989Z level=error msg="Unable to start controller" controller=gateway-api error="failed to wait for referencegrant caches to sync: timed out waiting for cache to be synced" subsys=gateway-controller
```

Signed-off-by: Tam Mach <tam.mach@cilium.io>
